src/main/java/org/terasology/thirst/ThirstAuthoritySystem.java

```diff
@@ -223,15 +224,16 @@ public void usedItem(DrinkConsumedEvent event, EntityRef item) {
      */
     @ReceiveEvent(components = {ThirstComponent.class})
     public void characterMoved(CharacterMoveInputEvent event, EntityRef character, ThirstComponent thirst) {
-        final float expectedDecay = event.isRunning() ? thirst.sprintDecayPerSecond : thirst.normalDecayPerSecond;
+        expectedDecay = event.isRunning() ? thirst.sprintDecayPerSecond : thirst.normalDecayPerSecond;
```

Comment:
Is `event.isRunning()` ever false? I think running is always active at the moment, so this should basically be a constant value? 🧐 After all, reacting to the `CharacterMoveInputEvent` (i.e., the _input_ event of the player hitting a button 🙄 ) might be the wrong level of abstraction.

Instead, the thirst system itself could react to the `AffectThirstEvent` and modify the value depending on the movement movement mode (or any other information taken from some component).

Reply:
event.isRunning() returns 1 while running and 0 while walking I guess .....(that's what the code seems to rely on)..